### PR TITLE
Fixes new tab page Brave Ads are not confirmed with the server - Only affects 1.18.x

### PR DESCRIPTION
**Pull request by tmancey**
Uplift of ... (This issue only affects 1.18.x) as new tab page ads was refactored in 1.19.x which is working as expected
Resolves https://github.com/brave/brave-browser/issues/12818

- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR.
- [ ] You have tested your change on Nightly. 
- [x] The PR milestones match the branch they are landing to. 

After you merge: 
- [ ] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/ad_events/new_tab_page_ads/new_tab_page_ad_event_viewed.cc
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/ad_events/new_tab_page_ads/new_tab_page_ad_event_viewed.cc:37]: Due to `NewTabPAgeAdEventViewed` being short-lived, we need to make a copy of `ads_` otherwise this member variable will not be captured

**Review by moritzhaller — APPROVED** [review]
LGTM

**Review by yachtcaptain23 — APPROVED** [review]

**Comment by kjozwiak** [comment]
Restarting `macOS` due to `build` failing as per https://ci.brave.com/job/pr-brave-browser-issues-12818-macos/5/execution/node/204/log/.

```
21:51:44  FAILED: obj/brave/common/extensions/api/generated_api_registration/generated_api_registration.o 
```

**Comment by kjozwiak** [comment]
`test-browser` failed on `macOS` due to several reward tests failing as per https://ci.brave.com/job/pr-brave-browser-issues-12818-macos/7/execution/node/251/log/. However, these are known intermittent time outs/failures and don't block uplift.

```
00:53:23  1 test failed:
00:53:23      RewardsPromotionBrowserTest.PromotionRemovedFromEndpoint (../../brave/components/brave_rewards/browser/test/rewards_promotion_browsertest.cc:230)
00:53:23  2 tests timed out:
00:53:23      RewardsBrowserTest.BackupRestoreModalHasNotice (../../brave/components/brave_rewards/browser/test/rewards_browsertest.cc:303)
00:53:23      RewardsPromotionBrowserTest.ClaimViaSettingsPage (../../brave/components/brave_rewards/browser/test/rewards_promotion_browsertest.cc:195)
```

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.18.x` approved after deliberating with @brave/uplift-approvers.

**Pull request merged**
